p2p/README.md

```diff
@@ -67,105 +66,431 @@ flowchart TD
+  +ExportedField
+  -unexportedField
+  +ExportedMethod(argType) returnType
+  -unexportedMethod()
+}
+
+class InterfaceType {
+    <<interface>>
+    +Method(argType) (returnType1, returnType2)
+}
+
+ConcreteType --|> InterfaceType : Interface realization
+
+ConcreteType --> OtherType : Direct usage
+ConcreteType --o OtherType : Composition
+ConcreteType --* OtherType : Aggregatation
+ConcreteType ..*  "(cardinality)" OtherType : Indirect (via interface)
+```
+
+### P2P Module / Router Decoupling
 
-### Message Propagation
+The P2P module encapsulates the `RaiTreeRouter` and `BackgroundRouter` submodules.
+The P2P module internally refers to these as the `stakedActorRouter` and `unstakedActorRouter`, respectively.
 
-Given `Local P2P Module` has a message that it needs to propagate:
+Depending on the necessary routing scheme (unicast / broadcast) and whether the peers involved are staked actors, a node will use one or both of these routers.
 
-<ul style="list-style-type: none;">
-    <li>1a. <code>Raintree Router</code> selects targets from the <code>Pokt Peerstore</code>, <strong>which only includes staked actors</strong></li>
-    <li>1b. <code>Background Router</code> selects targets from the libp2p <code>Peerstore</code>, <strong>which includes all P2P participants</strong></li>
-    <li>2. Libp2p <code>Host</code> manages opening and closing streams to targeted peers</li>
-    <li>3. <code>Remote P2P module</code>'s (i.e. receiver's) <code>handleStream</code> is called (having been registered via <code>setStreamHandler()</code>)</li>
-    <li>4a. <code>handleStream</code> propagates message via <code>Raintree Router</code></li>
-    <li>4b. <code>handleStream</code> propagates message via <code>Background Router</code></li>
-    <li>5a. Repeat step 1a from <code>Remote P2P Module</code>'s perspective targeting its next peers</li>
-    <li>5b. Repeat step 1b from <code>Remote P2P Module</code>'s perspective targeting its next peers</li>
-</ul>
+**Unicast**
+
+| Sender         | Receiver       | Router          | Example Usage                                        |
+|----------------|----------------|-----------------|------------------------------------------------------|
+| Staked Actor   | Staked Actor   | Raintree only   | Consensus (state sync) messages (to validators only) |
+| Unstaked Actor | Staked Actor   | Background only | Consensus (state sync) messages (to validators only) |
+| Unstaked Actor | Unstaked Actor | Background only | Consensus (state sync) & Debug (CLI) messages        |
+
+**Broadcast**
+
+| Broadcaster    | Receiver       | Router                | Example Usage                              |
+|----------------|----------------|-----------------------|--------------------------------------------|
+| Staked Actor   | Staked Actor   | Raintree + Background | Utility tx messages                        |
+| Unstaked Actor | Staked Actor   | Background only       | Utility tx messages (gossipsub redundancy) |
```

Comment:
```suggestion
| Unstaked Actor | Staked Actor   | Background only       | Utility tx messages (raintree redundancy) |
```

Reply:
I was attempting to make a distinction here between the "raintree redundancy layer" (as per #85) and the redundancy provided by gossipsub, out of band with respect to raintree's protocol. Do you have a suggestion for how to better articulate this distinction, perhaps more definitions?

Reply:
Got it. Let's go with `libp2p gossipub redundancy` then - closer to what you had before